AsyncDisplayKit/Layout/ASLayout.mm

```diff
@@ -88,6 +94,14 @@ - (instancetype)initWithLayoutElement:(id<ASLayoutElement>)layoutElement
 
     _sublayouts = sublayouts != nil ? [sublayouts copy] : @[];
     _flattened = NO;
+    
+    // Add sublayouts layout elements to an internal array to retain it while the layout lives
+    if (sublayouts.count > 0) {
```

Comment:
We could do some optimization here to only retain the sub layout elements if the layout element type is a node:

```objc
...
if (_layoutElementType == ASLayoutElementTypeDisplayNode && sublayouts.count > 0) {
...
```